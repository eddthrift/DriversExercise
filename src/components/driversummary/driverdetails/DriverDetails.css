.driver-details-container {
    background-color: white;
    padding-left: 10px;
}

.driver-details-item {
    background-color: white;
}

.driver-details-text {
    text-align: left;
    width: 80%;
    display: flex; 
    justify-content: space-between;
}