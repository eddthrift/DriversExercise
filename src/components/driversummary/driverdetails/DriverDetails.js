import { Grid, Stack, Typography } from '@mui/material';
import {GetActivityDuration} from '../../../utils/DriverUtils'
import React, {useState, useEffect} from 'react';
import DriverActivityBar from '../driveractivitybar/DriverActivityBar';

import './DriverDetails.css';

function DriverDetails({driver}){
    let [availableDuration, setAvailableDuration] = useState(0);
    let [driveDuration, setDriveDuration] = useState(0);
    let [workDuration, setWorkDuration] = useState(0);
    let [restDuration, setRestDuration] = useState(0);
    let [totalDuration, setTotalDuration] = useState(0);


    useEffect(() => {
        setAvailableDuration(GetActivityDuration(driver, 'available'));
        setDriveDuration(GetActivityDuration(driver, 'drive'));
        setWorkDuration(GetActivityDuration(driver, 'work'));
        setRestDuration(GetActivityDuration(driver, 'rest'));
        setTotalDuration(GetActivityDuration(driver, 'all'));
    },[driver]);

    return(
        <Grid container spacing={0} alignItems="center" className='driver-details-container'>
            <Grid item xs={3} className='driver-details-item'>
                <Stack direction={'row'} className='driver-details-text'>
                    <Typography>{driver.forename} {driver.surname}</Typography>
                    <Typography>{driver.vehicleRegistration}</Typography>
                </Stack>
            </Grid>

            <Grid item xs={1} className='driver-details-item'>
                <Typography>{availableDuration} min</Typography>
            </Grid>
            <Grid item xs={1} className='driver-details-item'>
                <Typography>{driveDuration} min</Typography>
            </Grid>
            <Grid item xs={1} className='driver-details-item'>
                <Typography>{workDuration} min</Typography>
            </Grid>
            <Grid item xs={1} className='driver-details-item'>
                <Typography>{restDuration} min</Typography>
            </Grid>
            <Grid item xs={1} className='driver-details-item'>
                <Typography>{totalDuration} min</Typography>
            </Grid>
            <Grid item xs={4} 
                className='driver-details-item'
                >
                <DriverActivityBar driver={driver}/>
            </Grid>
        </Grid>
    );
};

export default DriverDetails;