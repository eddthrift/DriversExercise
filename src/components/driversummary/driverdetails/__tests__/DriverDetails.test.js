import React from 'react';
import renderer from 'react-test-renderer';

import DriverDetails from '../DriverDetails';

var driver = {
    "driverID":51256058,
    "surname":"Wright",
    "forename":"Jonathon",
    "vehicleRegistration":"EB16FXW",
    "traces":[
        {
            "date":"2021-02-01",
            "activity":[
                {
                "startTime":"13:00",
                "type":"available",
                "duration":10
                },
                {
                "startTime":"13:10",
                "type":"drive",
                "duration":13
                },
                {
                "startTime":"13:23",
                "type":"drive",
                "duration":5
                },
            ]
        },
    ]
};

it('renders correctly', () => {
    const tree = renderer.create(<DriverDetails driver={driver}/>).toJSON();
    expect(tree).toMatchSnapshot();
});