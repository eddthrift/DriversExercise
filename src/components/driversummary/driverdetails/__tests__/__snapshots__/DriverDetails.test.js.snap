// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container driver-details-container css-1vam7s3-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 driver-details-item css-1equabv-MuiGrid-root"
  >
    <div
      className="MuiStack-root driver-details-text css-m69qwo-MuiStack-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Jonathon
         
        Wright
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        EB16FXW
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 driver-details-item css-u78q51-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      0
       min
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 driver-details-item css-u78q51-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      0
       min
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 driver-details-item css-u78q51-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      0
       min
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 driver-details-item css-u78q51-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      0
       min
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 driver-details-item css-u78q51-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      0
       min
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 driver-details-item css-gj1fbr-MuiGrid-root"
  >
    <div
      className="MuiStack-root css-1xcvc1q-MuiStack-root"
    >
      <div
        className="activity-box-idle MuiBox-root css-b7rhyr"
      />
      <div
        className="activity-box-idle MuiBox-root css-b7rhyr"
      />
      <div
        className="activity-box-idle MuiBox-root css-b7rhyr"
      />
      <div
        className="activity-box-idle MuiBox-root css-b7rhyr"
      />
      <div
        className="activity-box-idle MuiBox-root css-b7rhyr"
      />
      <div
        className="activity-box-idle MuiBox-root css-b7rhyr"
      />
      <div
        className="activity-box-idle MuiBox-root css-b7rhyr"
      />
    </div>
  </div>
</div>
`;
